### PR TITLE
Fix a race condition where we add a role and check for it immediately

### PR DESCRIPTION
**Pull request by stu-gott**
Fixes a race condition in a functional test.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by stu-gott** [comment]
/cc @jean-edouard 
Would you mind reviewing?

File: tests/vm_test.go [hunk 1]
```diff
@@ -1546,9 +1546,11 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
...
 
 					By("Checking VM creation permission using k8s client binary")
-					stdOut, _, err := tests.RunCommand(k8sClient, "auth", "can-i", "create", "vms", "--as", testUser)
-					Expect(err).ToNot(HaveOccurred())
-					Expect(strings.TrimSpace(stdOut)).To(Equal("yes"))
+					// It might take time for the role to propagate
+					Eventually(func() string {
+						stdOut, _, _ := tests.RunCommand(k8sClient, "auth", "can-i", "create", "vms", "--as", testUser)
+						return strings.TrimSpace(stdOut)
+					}, 120*time.Second, 10*time.Second).Should(Equal("yes"), fmt.Sprintf("test account '%s' was never granted permission to create a VM", testUser))
```
Comment (rmohr) [inline tests/vm_test.go:1553]: I would expect that we retry each second and wait up to 10 seconds. I am not so fluent in this area: What 3xactly need time to be propagated?

Comment (stu-gott) [reply tests/vm_test.go:1553]: The BeforeEach of this test is essentially running "oc adm policy add-role-to-user admin testuser". This does not appear to take effect instantly.

Your approach to timing is probably more reasonable.

**Comment by rmohr** [comment]
Will take the PR in once I am convinced by CI runs.

**Comment by rmohr** [comment]
Test results are good enough. Merging to increase test stability.

**Pull request merged**